src/validators/TupleValidator.ts

```diff
@@ -0,0 +1,41 @@
+import type { BaseError, IConstraint } from '..';
+import { BaseValidator } from './imports';
+import { Result } from '../lib/Result';
+import { ValidationError } from '../lib/errors/ValidationError';
+import { CombinedPropertyError } from '../lib/errors/CombinedPropertyError';
+
+export class TupleValidator<T extends BaseValidator<unknown>> extends BaseValidator<T[]> {
```

Comment:
This is entirely wrong, please read `UnionValidator`'s type.

Reply:
`UnionValidator`'s type is also wrong. Try to use it. 

1. It only allows rest params not an array.
2. It allows only 1 type of `BaseValidator` means u can use `union([s.string, s.number])`

I've sent a ss in discord server. Please check

Reply:
@kyranet what's the latest on this?

Reply:
I'll check it when I have time.